src/server/game/Movement/MovementGenerators/FleeingMovementGenerator.cpp

```diff
@@ -176,6 +176,7 @@ void FleeingMovementGenerator<T>::SetTargetLocation(T* owner)
     Movement::MoveSplineInit init(owner);
     init.MovebyPath(_path->GetPath());
     init.SetWalk(false);
+    init.SetVelocity(owner->GetSpeed(MOVE_RUN) * 0.66f);
```

Comment:
This movement generator is also used for regular fears (including targeting players), don't set it unconditionally

Reply:
I have to mention that some fear spells have aura effect that reduces speed, Intimidating Shout for instance. We may not need to change the flee speed manually.

Reply:
Maybe check unit_fleeing or fear aura count ?

Reply:
Maybe add a fear flag in `Unit::SetFeared` ?

Reply:
How about adding another arg to `MotionMaster::MoveFleeing`? `bool reducedSpeed`, set to true if inside `Unit::SetFeared` there are no fear auras present (its already checked there for other reasons) and in SAI flee action

Reply:
I just did that https://github.com/TrinityTBC/core/commit/00df0e39a7ff400463b7ec96716c4009ca03a1c7, lol